### PR TITLE
Ide 4587

### PR DESCRIPTION
**Pull request by jtydhr88**

### CODE REVIEW

**Comment by yanan** [comment]
Gradle-5.0-bin.zip file isn't fully downloaded. it retest

**Comment by yanan** [comment]
it retest

File: tools/plugins/com.liferay.ide.server.tomcat.core/src/com/liferay/ide/server/tomcat/core/PortalTomcatBundle.java [hunk 1]
```diff
@@ -261,6 +263,15 @@ protected int getDefaultJMXRemotePort() {
 
 		args.add("-Dcatalina.base=\"" + bundlePath.toPortableString() + "\"");
 		args.add("-Dcatalina.home=\"" + bundlePath.toPortableString() + "\"");
+
+		Version portalVersion = new Version(getVersion());
+
+		if (portalVersion.compareTo(new Version("7.0.0")) > 0) {
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.server.tomcat.core/src/com/liferay/ide/server/tomcat/core/PortalTomcatBundle.java]: hey @simonjhy  where do these properties come from?

Comment (gamerson) [reply tools/plugins/com.liferay.ide.server.tomcat.core/src/com/liferay/ide/server/tomcat/core/PortalTomcatBundle.java]: Can you create a new ticket and describe what problem this is solving and then send a separate PR ?  I'm not going to merge these changes in this PR.  Please go ahead and send a followup PR for this.

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/action/WatchWorkspaceModulesAction.java [hunk 2]
```diff
@@ -123,17 +128,34 @@ else if ("stop".equals(_action)) {
...
+						).map(
+							childProject -> LiferayCore.create(IBundleProject.class, childProject)
+						).filter(
+							Objects::nonNull
+						).parallel(
+						).forEach(
+							bundleProject -> {
+								IProject childProject = bundleProject.getProject();
+
+								new StopWatchBundleJob(
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/action/WatchWorkspaceModulesAction.java]: Let's not create a single stop watch bundle job for each project.  But instead let's collect all of the projects that need to have the bundle uninstalled and pass to the stop watch bundle job instead.

File: tools/plugins/com.liferay.ide.server.tomcat.core/src/com/liferay/ide/server/tomcat/core/PortalTomcatBundle.java (hunk 1 above)
Comment (simonjhy) [reply tools/plugins/com.liferay.ide.server.tomcat.core/src/com/liferay/ide/server/tomcat/core/PortalTomcatBundle.java]: this properties get from portal-impl.jar.the method getVersion is existed method of portalBundle class

File: tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/action/WatchWorkspaceModulesAction.java (hunk 2 above)
Comment (simonjhy) [reply tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/action/WatchWorkspaceModulesAction.java]: ok

File: tools/plugins/com.liferay.ide.server.tomcat.core/src/com/liferay/ide/server/tomcat/core/PortalTomcatBundle.java (hunk 1 above)
Comment (simonjhy) [reply tools/plugins/com.liferay.ide.server.tomcat.core/src/com/liferay/ide/server/tomcat/core/PortalTomcatBundle.java]: these three new arguments are come from tomcat 9.0.1 original parameters.

Comment (gamerson) [reply tools/plugins/com.liferay.ide.server.tomcat.core/src/com/liferay/ide/server/tomcat/core/PortalTomcatBundle.java]: Got it.  Thanks for checking that.  Please those in a new update.

Comment (simonjhy) [reply tools/plugins/com.liferay.ide.server.tomcat.core/src/com/liferay/ide/server/tomcat/core/PortalTomcatBundle.java]: ok

**Pull request closed without merging**